### PR TITLE
ER-920 Mobile side menu

### PR DESCRIPTION
**Pull request by martikat**
[ER-920](https://dfedigital.atlassian.net/browse/ER-920)

This also includes the fix for [ER-1009](https://dfedigital.atlassian.net.mcas.ms/browse/ER-1009)

### CODE REVIEW

**Comment by peterdavidhamilton** [comment]
@martikat 

- [ ] entire stylesheets from the old design are still present and should be removed
- [ ] the active links aren't needed and the burger menu is always visible so the use of a conditional is not appropriate for the new design
- [ ] if there is a JS issue for the accordion action on the nav links let's confirm that on a prod deployment

File: app/models/page.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def breadcrumbs
     list = [self, self&.parent, self&.parent&.parent, self&.parent&.parent&.parent].compact.reverse
     list.shift
     list.each_with_object({ 'Home' => '/' }) do |obj, memo|
-      memo[obj.title] = obj.path
```
Comment (peterdavidhamilton) [inline app/models/page.rb:48]: This code looks like `#footer_path`, is there a DRY-er way of doing this?

File: app/components/header_component.html.slim [hunk 2]
```diff
@@ -7,13 +7,19 @@
...
       = link_to service_name, service_url, class: 'dfe-header__link dfe-header__link--service'
     #content-header.dfe-header__content.dfe-header-f-dtonly.dfe-header-f-toplinks
-      .dfe-header-f-top-links
-        / ul.dfe-header__action-links
-        /   - action_links.each do |item|
-        /     = item
       .dfe-header__menu
         button#toggle-menu.dfe-header__menu-toggle aria-controls='header-navigation' aria-expanded='false' data-action='click->reveal#toggle'
           | Menu
+    / This logic has been commented out as it is not required currently
```
Comment (peterdavidhamilton) [inline app/components/header_component.html.slim:13]: Could you check that this way of commenting doesn't result in visible HTML comments and use an alternative approach if it does please. Would a format like the following separated by blank lines read any better.

```

-# original header template
<their code>

-# custom service template
<our code>

```

File: app/assets/stylesheets/metadata.scss [hunk 3]
```diff
@@ -201,4 +201,180 @@ dl.gem-c-metadata__list {
   padding: 15px 10px 15px 20px;
   margin-top: 60px;
   border-left: 5px solid #347ca9;
-}
\ No newline at end of file
+}
```
Comment (peterdavidhamilton) [inline app/assets/stylesheets/metadata.scss:204]: If the entire contents of this file is responsible for styling the `citation` partial, let's call it that. Anything that is unrelated should be moved out.

File: app/assets/stylesheets/links.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.govuk-breadcrumbs li {
```
Comment (peterdavidhamilton) [inline app/assets/stylesheets/links.scss:1]: This file is specifically about the breadcrumb styling.

File: app/assets/stylesheets/other-resources.scss [hunk 5]
```diff
@@ -0,0 +1,99 @@
+.hf-other {
+  padding-top: govuk-spacing(9);
+  padding-bottom: govuk-spacing(9);
+}
+
+.hf-ctas {
```
Comment (peterdavidhamilton) [inline app/assets/stylesheets/other-resources.scss:6]: I see this class used only once in the CTA view partial.

File: app/assets/stylesheets/application.scss [hunk 6]
```diff
@@ -25,24 +25,19 @@ $govuk-page-width: 1200px;
...
-@import '_cookie-banner';
-@import '_links';
-@import '_flash';
+@import 'print';
+@import 'card';
+@import 'side-nav';
+@import 'other-resources';
+@import 'cta';
 
+@import 'subnav';
```
Comment (peterdavidhamilton) [inline app/assets/stylesheets/application.scss:39]: This file has styling (that isn't imported) which can still be tidied up and removed. `app-sidebar` for example.

**Review by peterdavidhamilton — CHANGES_REQUESTED** [review]

File: app/components/header_component.html.slim (hunk 2 above)
Comment (martikat) [reply app/components/header_component.html.slim:13]: This comment isn't shown when inspecting where it is used. I will update this so it is more readable though 👍

File: app/assets/stylesheets/subnav.scss [hunk 7]
```diff
@@ -32,12 +32,12 @@
 
 .app-subnav--mobile {
```
[the remark this answers is not in the document]
Comment (martikat) [reply app/assets/stylesheets/subnav.scss:33]: It looks like this isn't being used. I will go ahead and remove this

File: app/assets/stylesheets/application.scss (hunk 6 above)
Comment (martikat) [reply app/assets/stylesheets/application.scss:39]: I have removed subnav in the next commit

File: app/components/header_component.html.slim [hunk 8]
```diff
@@ -6,11 +6,7 @@
         = image_tag 'dfe-logo-alt.png', class: 'dfe-logo-hover', alt: 'DfE homepage'
     #service-name
       = link_to service_name, service_url, class: 'dfe-header__link dfe-header__link--service'
-    #content-header.dfe-header__content.dfe-header-f-dtonly.dfe-header-f-toplinks
-      .dfe-header__menu
-        button#toggle-menu.dfe-header__menu-toggle aria-controls='header-navigation' aria-expanded='false' data-action='click->reveal#toggle'
-          | Menu
-    / This logic has been commented out as it is not required currently
+    / Original header template
```
Comment (peterdavidhamilton) [inline app/components/header_component.html.slim:9]: `/!` this is the syntax I was concerned about to avoid making it visible the browser. I would recommend leveraging the single `/` where possible when it is all indented.

Comment (martikat) [reply app/components/header_component.html.slim:9]: I have updated this in the next commit to use a single `/` for the comment

File: app/assets/stylesheets/cta.scss [hunk 9]
```diff
@@ -15,6 +15,26 @@
...
+    padding: 20px;
+  }
+
+  .table-scroll {
+    overflow: scroll;
+  }
+}
+
 @media (max-width: 40.0525em) {
   .hf-card {
```
Comment (peterdavidhamilton) [inline app/assets/stylesheets/cta.scss:39]: 👀

File: app/assets/stylesheets/downloadable-media.scss [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+  -moz-osx-font-smoothing: grayscale;
+  font-weight: 400;
+  font-size: 16px;
+  font-size: 1rem;
+  line-height: 1.25;
+  margin: 0 0 15px;
+  color: #505a5f
+}
+
+@media print {
```
Comment (peterdavidhamilton) [inline app/assets/stylesheets/downloadable-media.scss:138]: Please combine the `print` media block definitions.

**Review by peterdavidhamilton — CHANGES_REQUESTED** [review]
This [npm package](https://github.com/alphagov/stylelint-config-gds) replaces [this rubygem](https://github.com/alphagov/scss-lint-govuk). Can you add it and see what it says?

File: app/assets/stylesheets/cta.scss (hunk 9 above)
Comment (martikat) [reply app/assets/stylesheets/cta.scss:39]: This can be deleted (it is not being used) 👍  I will include this change in the next commit

File: app/models/page.rb (hunk 1 above)
Comment (peterdavidhamilton) [reply app/models/page.rb:48]: I have addressed this in https://github.com/DFE-Digital/early-years-foundation-reform/pull/781

**Review by peterdavidhamilton — APPROVED** [review]
